### PR TITLE
Execute unkown mojos by default

### PR DESCRIPTION
**Pull request by laeubi**
See 
- https://github.com/eclipse-m2e/m2e-core/issues/1013

### CODE REVIEW

**Review by HannesWell — COMMENTED** [review]
I did not check it great depth but the change looks good in general. Only a few suggestions for clean-ups or text improvements are below.

Did you check that plug-ins like the `maven-compiler-plugin` are not executed? If it is, we should add a corresponding mapping to ignore it.

And maybe I oversaw this, but are warnings are really issued when that setting is selected?
The handling should probably happen in `LifecycleMappingFactory.instantiateProjectConfigurators()`.

It would be good to extract the clean-ups/other minor fixes into a first commit to make the actual change more focused. The additional type checks could go in there. They might not have been necessary before, but it would have been cleaner code, so I think that could go in a first commit. An extra PR is IMO not necessary.

**Comment by laeubi** [comment]
According to the mappings e.g. here:


there is an explicit configuration for the (default) compile already, this change only applies to mojos that previously would have been marked as error by m2e because of missing connectors.

File: org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/ProjectRegistryManager.java [hunk 1]
```diff
@@ -649,10 +649,12 @@ private void detachMappingSources(Map<MojoExecutionKey, List<IPluginExecutionMet
...
-          PluginExecutionMetadata execution = (PluginExecutionMetadata) iterator.next();
-          execution = execution.clone();
-          execution.setSource(null);
-          iterator.set(execution);
+          IPluginExecutionMetadata metadata = iterator.next();
+          if(metadata instanceof PluginExecutionMetadata execution) {
+            execution = execution.clone();
+            execution.setSource(null);
+            iterator.set(execution);
+          }
```
[the remark this answers is not in the document]
Comment (laeubi) [reply org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/ProjectRegistryManager.java:657]: I'll keep this for now as I actually not changed the code inside, not sure if this is required for anything...

**Comment by HannesWell** [comment]
> there is an explicit configuration for the (default) compile already, this change only applies to mojos that previously would have been marked as error by m2e because of missing connectors.

OK, thanks for the clarification.

> Tests should now pass as well..

There are still test-failures, at least in the GH-workflow. I re-triggerd those workflows to see if they are permanent. If they are, can you please fix them?

**Comment by HannesWell** [comment]
> > Tests should now pass as well..
> 
> There are still test-failures, at least in the GH-workflow. I re-triggerd those workflows to see if they are permanent. If they are, can you please fix them?

The tests are still failing.

**Comment by laeubi** [comment]
I think last time the issue was that GH do not update the modules ...

**Comment by HannesWell** [comment]
> I think last time the issue was that GH do not update the modules ...

https://github.com/eclipse-m2e/m2e-core/pull/1203 fix that.

**Pull request merged**